tasks/ast_tools/src/generators/typescript.rs

```diff
@@ -224,7 +224,8 @@ fn generate_ts_type_def_for_struct_field<'s>(
             }
         }
 
-        if field_type_name.contains('|') {
+        // need `type` instead of `interface` when flattening BindingPattern
+        if field_type_name.contains('|') || field_type_name == "BindingPattern" {
```

Comment:
Hard-coding `BindingPattern` is bad, but I didn't find a good way to force `output_as_type`. I can look into supporting this case in general later.

Reply:
Yes, not ideal in long term, but totally fine for now. Personally, I think we should look at changing the Rust types for `BindingPattern` anyway. It's a bit weird compared to all the other AST types - it's the only node which doesn't have it's own `Span` but gets its span from a child, for example.